test/jdk/java/lang/invoke/indify/Indify.java

```diff
@@ -253,28 +360,32 @@ else if (!keepgoing)
 
     private void ensureDirectory(File dir) {
         if (dir.mkdirs() && !quiet)
-            System.err.println("created "+dir);
+            System.err.println("Created new directory to: "+dir);
     }
 
     public void indifyFile(File f, File dest) throws IOException {
-        if (verbose)  System.err.println("reading "+f);
-        ClassFile cf = new ClassFile(f);
-        Logic logic = new Logic(cf);
-        boolean changed = logic.transform();
+        if (verbose)  System.err.println("reading " + f);
+        ClassModel model = parseClassFile(f);
+        Logic logic = new Logic(model);
+        Boolean changed = logic.transform();
         logic.reportPatternMethods(quiet, keepgoing);
         if (changed || all) {
             File outfile;
             if (dest != null) {
                 ensureDirectory(dest);
-                outfile = classPathFile(dest, cf.nameString());
+                outfile = classPathFile(dest, model.thisClass().name().stringValue());
             } else {
                 outfile = f;  // overwrite input file, no matter where it is
             }
-            cf.writeTo(outfile);
-            if (!quiet)  System.err.println("wrote "+outfile);
+            Files.write(outfile.toPath(), transformToBytes(logic.classModel));
+            if (!quiet) System.err.println("wrote "+outfile);
         }
     }
 
+    byte[] transformToBytes(ClassModel classModel) {
+        return of().transform(classModel, ClassTransform.ACCEPT_ALL);
```

Comment:
What is the purpose of this transformation?

Reply:
Transforming the classModel back to the bytes array in order to write the transformed classfile

Reply:
And what is the purpose of parsing a model from bytes and transforming it back to bytes without a change?

Reply:
This transformation is called only after the `classModel` is transformed: 
- at line 472  and 380 when the `transformToBytes()` method is called is after effectively transforming the `classModel`

Reply:
There isn’t currently any API for converting a `ClassModel` directly to a `byte[]` without calling a `ClassFile​::transformClass(…)` method.

Reply:
Repeated parsing (bytes -> model) and transforming back (model -> bytes) is an anti-pattern. Indify tool should operate as a single-pass transformation. Looping on a class model methods / instructions while replacing the class model itself (with something partially transformed) is not the right approach.